### ISSUE TITLE
Foreign key creation does not trigger adding-foreign-key-constraint rule failure when part of CREATE TABLE

### ISSUE THREAD

**Issue by luisgc93** [sbdchd/squawk#938]
First of all, thanks for creating such an amazing tool. I've used it in different teams already and it's an excellent way of adding automated guardrails when using PostgreSQL and other databases in production.

The other day, I noticed the following statement did not trigger a job failure:
`````
CREATE TABLE table_name (
    id BIGSERIAL NOT NULL, 
    my_fk_id INTEGER NOT NULL, 
    FOREIGN KEY(my_fk_id) REFERENCES other_table (id) ON DELETE CASCADE
);
UPDATE alembic_version SET version_num='xyz' WHERE alembic_version.version_num = 'abc';
COMMIT;
`````

The psql docs indicate that adding an FK to a new table still requires a SHARE ROW EXCLUSIVE on the referenced table:
https://www.postgresql.org/docs/current/sql-createtable.html#SQL-CREATETABLE-PARMS-REFERENCES

```
The addition of a foreign key constraint requires a SHARE ROW EXCLUSIVE lock on the referenced table
```
Based on this,  I assumed it would also make the `adding-foreign-key-constraint` rule fail. Alternatively, it could be configured by a new separate rule e.g. `adding-foreign-key-constraint-in-create-table`

**Comment by sbdchd** [comment]
Yeah in this case, table `x` would be empty but we need to get a lock on table `y`

So it ends up being an issue with table `y`. Once it has the lock, then it doesn't need to check anything since table `x` is empty.


To reproduce:


```sql
-- setup
CREATE TABLE y (
    id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
    value BIGINT
);
INSERT INTO y (value)
SELECT gs
FROM generate_series(1, 10000000) AS gs;


-- connection a
begin;
insert into y(value) values (1);

-- meanwhile in connection b,

begin;
-- this hangs since it can't get the lock
CREATE TABLE x (
  id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
  fk BIGINT NOT NULL REFERENCES y (id)
);
commit;
```
